src/modules/notifications/notifications.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable @typescript-eslint/indent */
+import Mailgun from 'mailgun-js';
+import Email from '../../lib/emails';
+
+/**
+ * @description Send out email notificaions from the subList
+ * @param {Array<string>} subList A list of subscribers
+ * @param {Date} deliveryTime Time that the email will be delivered
+ */
+const notifyMany = async (
+    subList: Array<string>,
+    deliveryTime: Date
+): Promise<Array<string | Mailgun.messages.SendResponse>> => {
+    if (subList.length === 0) {
+        throw new Error('Empty subscribe list');
+    }
+    const results: Array<Promise<string | Mailgun.messages.SendResponse>> = [];
+    const subsetSize = 999;
+    for (let i = 0; i < subList.length; i += subsetSize) {
+        // Take max of 1k emails to send for each mailgun batch
+        const emails = subList.slice(i, i + subsetSize);
```

Comment:
Is this bounded correctly? Where N = iteration count, N = 2, i = 999, i  + subsetSize = 1998  -- could do something like `Math.min(subList.length - 1, i + subSetSize)`